thetatauCMT/chapters/models.py

```diff
@@ -516,7 +518,121 @@ def get_generic_chapter_emails(self):
             self.email_treasurer,
             self.email_corresponding_secretary,
             self.email,
-        ]
+        ]   
+
+    def get_current_and_future(self):
+        # list all officers that currently hold an executive board position
+        # current and future
+        officers = self.members.filter(
+            roles__role__in=CHAPTER_OFFICER,
+            roles__end__gte=TODAY_END,
+        )
+        current_and_future_regent = officers.filter(roles__role="regent")
+        current_and_future_scribe = officers.filter(roles__role="scribe")
+        current_and_future_vice = officers.filter(roles__role="vice regent")
+        current_and_future_treasurer = officers.filter(roles__role="treasurer")
+        current_and_future_cosec = officers.filter(roles__role="corresponding secretary")
+        return current_and_future_regent,current_and_future_scribe,current_and_future_vice,current_and_future_treasurer,current_and_future_cosec
+
+    def get_previous_officers(self):
+        # list the most recent officer that held position
+        previous_officers = self.members.filter(
+                roles__role__in=CHAPTER_OFFICER,
+                roles__end__gte=TODAY_END - timedelta(30 * 8),# they ended their role in the last 8 months
+            )
+        past_regent = previous_officers.filter(roles__role="regent").order_by('roles__end').first()
+        past_scribe = previous_officers.filter(roles__role="scribe").order_by('roles__end').first()
+        past_vice = previous_officers.filter(roles__role="vice regent").order_by('roles__end').first()
+        past_treasurer = previous_officers.filter(roles__role="treasurer").order_by('roles__end').first()
+        past_cosec = previous_officers.filter(roles__role="corresponding secretary").order_by('roles__end').first()
+        
+        return past_regent,past_scribe,past_vice,past_treasurer,past_cosec
+
+
+    def get_about_expired_coucil(self):
+        pos = ['regent','scribe','vice regent','treasurer','corresponding secretary'] #list all the positions to check for and loop through
+        officers_to_update,officers_to_notify,current_and_future_officer,previous_officer,emails = [],[],[],[],[]
```

Comment:
You have so many lists and things to track and remember.
Use two. No more. officers_to_update and officers_to_notify